### PR TITLE
PR for Issue 21159: Nonce/request URL storage and OP discovery enhancements

### PR DESCRIPTION
**Pull request by ayoho**
- Update authentication dialog to use storage subsystem for nonce and original request URL values
- Jakarta authentication dialog discovery enhancements:
    - Adds a caching mechanism for OP metadata to avoid sending discovery requests too often
    - Adds the bulk of the remaining code to perform OP discovery
- Added code to Jakarta authentication dialog to redirect to the authorization endpoint with all appropriate request parameters included
    - Includes required parameters (scope, response type, client ID, redirect URI, and state)
    - Includes optional parameters depending on the OIDC client config (nonce, prompt, response mode, display, extra parameters)

For #21159

### CODE REVIEW

**Comment by ayoho** [comment]
Good personal build: https://wasrtc.hursley.ibm.com:9443/jazz/resource/itemOid/com.ibm.team.build.BuildResult/_ZE9WEB16Ee2b2O1gXXKb4w. Had child builds fail due to some infrastructure issues, but as far as I can tell all OIDC tests ran and passed.

File: dev/io.openliberty.security.oidcclientcore.internal/resources/io/openliberty/security/oidcclientcore/internal/resources/OidcClientCoreMessages.nlsprops [hunk 1]
```diff
@@ -29,10 +29,18 @@ URL_NOT_HTTPS=CWWKS2402E: The {0} endpoint URL that is configured for the {1} Op
 URL_NOT_HTTPS.explanation=The endpoint URL must use the HTTPS protocol to ensure that requests are secure.
 URL_NOT_HTTPS.useraction=Update the endpoint URL to use the HTTPS protocol.
 
-DISCOVERY_EXCEPTION=CWWKS2403E: The {0} OpenID Connect client encountered an error while discovering metadata for the OpenID Connect provider. {1}
+DISCOVERY_EXCEPTION=CWWKS2403E: The {0} OpenID Connect client encountered an error while discovering metadata for the OpenID Connect provider from the [{1}] URL. {2}
```
Comment (dmuelle) [inline dev/io.openliberty.security.oidcclientcore.internal/resources/io/openliberty/security/oidcclientcore/internal/resources/OidcClientCoreMessages.nlsprops]:
```suggestion
DISCOVERY_EXCEPTION=CWWKS2403E: The {0} OpenID Connect client encountered the following error during discovery of metadata for the OpenID Connect provider from the [{1}] URL: {2}
```

**Review by dmuelle — CHANGES_REQUESTED** [review]
ID review- let me know if any questions, thanks

File: dev/io.openliberty.security.oidcclientcore.internal/resources/io/openliberty/security/oidcclientcore/internal/resources/OidcClientCoreMessages.nlsprops (hunk 1 above)
Comment (ayoho) [reply dev/io.openliberty.security.oidcclientcore.internal/resources/io/openliberty/security/oidcclientcore/internal/resources/OidcClientCoreMessages.nlsprops]: Update pushed 👍

**Comment by ayoho** [comment]
#libby

**Review by dmuelle — APPROVED** [review]
LGTM, thanks

**Review by teddyjtorres — APPROVED** [review]
Thank you very much for these changes.

**Pull request merged**